xdsl/dialects/builtin.py

```diff
@@ -2197,9 +2197,17 @@ def create_dense_index(
     @staticmethod
     def create_dense_int(
         type: RankedStructure[_IntegerAttrType],
-        data: Sequence[int] | Sequence[IntegerAttr[_IntegerAttrType]],
+        data: int
+        | IntegerAttr[_IntegerAttrType]
+        | Sequence[int]
+        | Sequence[IntegerAttr[_IntegerAttrType]],
     ) -> DenseIntOrFPElementsAttr[_IntegerAttrType]:
-        if len(data) and isinstance(data[0], IntegerAttr):
+        if isinstance(data, IntegerAttr):
+            data = data.value.data
+
+        if isinstance(data, int):
+            data = (data,) * prod(type.get_shape())
+        elif len(data) and isinstance(data[0], IntegerAttr):
```

Comment:
This feels weird to me. Could we not just get the bytes directly? Is there a way to avoid the roundtrip through int?

Reply:
It's possible there's a lot of better ways to do this, I wasn't really going for efficiency with this PR. I think by far the better optimisation would be normalising before duplicating the integer. 

By "avoid the roundtrip" do you mean just packing the input integer multiple times?

Reply:
I'm OK to defer this to a different PR, but here we have direct access to the bytes, so could just replicate those directly instead of first getting integer, and then packing again.
```
>>> b'beef' * 8
b'beefbeefbeefbeefbeefbeefbeefbeef'
```

Reply:
Sure, do you think you'll have the motivation to try this on top of the stack?

Reply:
I'll do it now

Reply:
#4393